File: readability/readability.c
#include <stdio.h>
#include <cs50.h>
#include <math.h>

void get_grade(string text);

int main(void)
{
    string text = get_string("Text: ");
    get_grade(text);
}

int count_letters(string text)
{
    int counter = 0;
    for (int i = 0; text[i] != 0; i++)
    {
        if ((text[i] >= 65 && text[i] <= 90) || (text[i] >= 97 && text[i] <= 122))
        {
            counter++;
        }
    }
    return counter;
}

int count_words(string text)
{
    int counter = 1;
    for (int i = 0; text[i] != 0; i++)
    {
        if (text[i] == 32)
        {
            counter++;
        }
    }
    return counter;
}

int count_sentences(string text)
{
    int counter = 0;
    for (int i = 0; text[i] != 0; i++)
    {
        if (text[i] == 33 || text[i] == 46 || text[i] == 63)
        {
            counter++;
        }
    }
    return counter;
}

float coleman_liau_index(string text)
{
    int letters = count_letters(text);
    int words = count_words(text);
    int sentences = count_sentences(text);

    float L = 100 * (float) letters / (float) words;
    float S = 100 * (float) sentences / (float) words;

    return 0.0588 * L - 0.296 * S - 15.8;
}

void get_grade(string text)
{
    float index = coleman_liau_index(text);
    if (index < 1)
    {
        printf("Before Grade 1\n");
    }
    else if (index >= 16)
    {
        printf("Grade 16+\n");
    }
    else
    {
        int grade = (int) roundf(index);
        printf("Grade %i\n", grade);
    }
}

File: pset6/readability/readability.py
def count_letters(text):
    counter = 0
    for i in text:
        if ((i >= "a" and i <= "z") or (i >= "A" and i <= "Z")):
            counter += 1
    return counter


def count_words(text):
    counter = 1
    for i in text:
        if (i == " "):
            counter += 1
    return counter


def count_sentences(text):
    counter = 0
    for i in text:
        if (i == "." or i == "!" or i == "?"):
            counter += 1
    return counter


def coleman_liau_index(text):
    letters = count_letters(text)
    words = count_words(text)
    sentences = count_sentences(text)
    L = 100 * letters / words
    S = 100 * sentences / words
    return 0.0588 * L - 0.296 * S - 15.8


def get_grade(text):
    index = coleman_liau_index(text)
    if (index < 1):
        print("Before Grade 1")
    elif (index >= 16):
        print("Grade 16+")
    else:
        grade = int(round(index))
        print(f"Grade {grade}")


text = input("Text: ")
get_grade(text)
